cli/command/deploy_test.go

```diff
@@ -0,0 +1,183 @@
+package command
+
+import (
+	"testing"
+
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/urfave/cli"
+)
+
+func TestCreateDeploy(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewDeployCommand(base.Command())
+
+	file, close := createTempFile(t, "dpl_file")
+	defer close()
+
+	req := models.CreateDeployRequest{
+		DeployName: "dpl_name",
+		DeployFile: []byte("dpl_file"),
+	}
+
+	base.Client.EXPECT().
+		CreateDeploy(req).
+		Return("jid", nil)
+
+	job := &models.Job{
+		Status: "Completed",
+		Result: "dpl_id",
+	}
+
+	base.Client.EXPECT().
+		ReadJob("jid").
+		Return(job, nil)
+
+	base.Client.EXPECT().
+		ReadDeploy("dpl_id").
+		Return(&models.Deploy{}, nil)
+
+	c := NewContext(t, []string{file.Name(), "dpl_name"}, nil)
+	if err := command.create(c); err != nil {
+		t.Fatal(err)
+	}
+}
```

Comment:
We should be folding in the functional patterns for testing the wait/no-wait config in this test.